x/oracle/keeper/oracle.go

```diff
@@ -42,6 +43,69 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 	return nil
 }
 
+func (k Keeper) ApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApproveOracleRegistration) error {
+
+	if err := k.validateApproveOracleRegistration(ctx, msg); err != nil {
+		return err
```

Comment:
I think if directly return like this, than client cannot see correctly log.

You must be used like `errors.go`

```go
ErrApproveOracleRegistration = sdkerrors.Register(ModuleName, 6, "error while approve ...")
```

Reply:
Or return `ErrApproveOracleRegistration` from the outermost layer (`msg_server_*.go`), if you think we don't need fine-grained errors for now. Unfortunately, there is no concrete policy about returning sentinel error, IIRC.

Reply:
Thanks, I fixed it.